### ISSUE TITLE
Discrepancy in evaluation results vs. paper

### ISSUE THREAD

**Issue by ricolxwz** [sanjayss34/lm-listener#14]
Hi, we've found that our evaluation results are inconsistent with the paper, with a significant gap. We used the official pre-trained model and code. 

Results: 

```
l2 0.65775126
fid 30.773574559764096
fid2 32.30244528068907
diversity 2.9099445
diversity GT 2.732136
var 0.10657453
var GT 0.11799331
```

Results in paper: 

```
l2 0.43
fid 18.22
var 0.116
diversity 2.81
```

Inference command: 

```sh
python train_t2m_trans_paper.py \
        --exp-name listener_trevor_paper_1030 \
        --batch-size 8 \
        --nb-code 256 \
        --drop-out-rate 0.1 \
        --resume-pth output/VQVAE-Paper/net_iter300000.pth \
	--vq-name VQVAE_trevor \
        --out-dir output \
        --total-iter 0 \
        --lr-scheduler 150000 \
        --lr 0.00005 \
        --dataname face_trevor \
        --down-t 3 \
        --depth 3 \
        --quantizer ema_reset \
        --eval-iter 2000 \
        --pkeep 0.50 \
        --dilation-growth-rate 3 \
        --vq-act relu \
        --max-motion-length 240 \
        --gpt2 gpt2-medium \
        --print_val_pred \
        --gradient_accumulation_steps 2 \
        --manual-bf16 \
        --delay-start-frames 96 \
        --max-time-before 3 \
        --save-name subdir \
        --seed 50
```

Evaluation command: 

```
python evaluate.py \
--output_dir output/listener_trevor_paper_1030 \
--segments_path dataset/trevor/segments_val.pth \
--mean_std_path dataset/trevor/
```

Identity: Trevor. 

Could you please help me find out how I can reproduce the result in the paper? Thank you very much.

**Comment by sanjayss34** [comment]
Hi thanks for your interest in our work, and I'm sorry for the delay. The results in Table 2 are computed on the test set. In order to run inference on the test set, you can use the `--test-eval` flag. When running evaluation on the test set, you should use the `segments_test.pth` file. I cloned a new copy of this repo and ran the evaluation on the Trevor test set with the provided pre-trained LM and VQ model checkpoints and obtained these results:
```
l2 0.44850695
fid 18.506493765599807
fid2 19.98447163771411
diversity 2.886863
diversity GT 2.5769398
var 0.12300264
var GT 0.11482989
```
I also ran the evaluation on the Trevor validation set and obtained these results:
```
l2 0.51665086
fid 22.1531881843784
fid2 23.665014062241728
diversity 2.6814933
diversity GT 2.7156777
var 0.102279924
var GT 0.11799331
```
This is the command I used to run inference on the test set (similar to yours):
```
python train_t2m_trans.py     --exp-name listener_trevor     --batch-size 8     --nb-code 256     --drop-out-rate 0.1     --resume-pth pretrained/VQVAE_l1smooth_c256_p8_datav2/net_iter300000.pth     --vq-name VQVAE_trevor     --out-dir output     --total-iter 0     --lr-scheduler 150000     --lr 0.00005     --dataname face_trevor     --down-t 3     --depth 3     --quantizer ema_reset     --eval-iter 2000     --pkeep 0.50     --dilation-growth-rate 3     --vq-act relu     --max-motion-length 240     --gpt2 gpt2-medium     --print_val_pred     --gradient_accumulation_steps 2     --delay-start-frames 96     --max-time-before 3     --save-name preds_trevor_test --test-eval --seed 50     --resume-trans pretrained/trevor_hist3/net_last.pth
```
